metrics/scaling/k8s_scale_rapid_nc.sh

```diff
@@ -0,0 +1,337 @@
+
+	local latency_json="$(cat << EOF
+			"latency_time": {
+				"Percentiles": [$(IFS=, ; echo "${latency_percentiles[*]}")],
+				"Result": ${latency_percentiles[$(( ${#latency_percentiles[@]} / 2 ))]},
+				"Units" : "ms"
+			}
+EOF
+	)"
+
+	metrics_json_add_array_fragment "$latency_json"
+
+	info "launch [$launch_time_ms]"
+
+	metrics_json_close_array_element
+}
+
+init() {
+	framework_init
+}
+
+save_config(){
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"testname": "${TEST_NAME}",
+		"NUM_PODS": ${NUM_PODS},
+		"STEP": ${STEP},
+		"wait_time": ${wait_time},
+		"delete_wait_time": ${delete_wait_time},
+		"settle_time": ${settle_time},
+		"nc_reqs": ${nc_reqs},
+		"nc_req_msg_len": ${nc_req_msg_len},
+		"nc_percentiles": [$(IFS=, ; echo "${nc_percentiles[*]}")]
+	}
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+run() {
+	info "Running test"
+
+	trap cleanup EXIT QUIT KILL
+
+	metrics_json_start_array
+
+	for reqs in $(seq ${STEP} ${STEP} ${NUM_PODS}); do
+		info "Testing replicas ${reqs} of ${NUM_PODS}"
+		# Generate the next yaml file
+
+		local runtime_command
+		if [ -n "$RUNTIME" ]; then
+			runtime_command="s|@RUNTIMECLASS@|${RUNTIME}|g"
+		else
+			runtime_command="/@RUNTIMECLASS@/d"
+		fi
+
+		local input_template
+		local generated_file
+		if [ "$use_api" != "no" ]; then
+			input_template=$input_json
+			generated_file=$generated_json
+		else
+			input_template=$input_yaml
+			generated_file=$generated_yaml
+		fi
+
+		sed -e "s|@REPLICAS@|${reqs}|g" \
+			-e $runtime_command \
+			-e "s|@DEPLOYMENT@|${deployment}|g" \
+			-e "s|@LABEL@|${LABEL}|g" \
+			-e "s|@LABELVALUE@|${LABELVALUE}|g" \
+			-e "s|@GRACE@|${grace}|g" \
+			-e "s#@PODCOMMAND@#${pod_command}#g" \
+			< ${input_template} > ${generated_file}
+
+		info "Applying changes"
+		local start_time=$(date +%s%N)
+		if [ "$use_api" != "no" ]; then
+			# If this is the first launch of the deploy, we need to use a different URL form.
+			if [ $reqs == ${STEP} ]; then
+				curl -s ${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments -XPOST -H 'Content-Type: application/json' -d@${generated_file} > /dev/null
+			else
+				curl -s ${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments/${deployment} -XPATCH -H 'Content-Type:application/strategic-merge-patch+json' -d@${generated_file} > /dev/null
+			fi
+		else
+			kubectl apply -f ${generated_file}
+		fi
+
+		kubectl rollout status --timeout=${wait_time}s deployment/${deployment}
+		local end_time=$(date +%s%N)
+		local total_milliseconds=$(( (end_time - start_time) / 1000000 ))
+		info "Took $total_milliseconds ms ($end_time - $start_time)"
+
+		sleep ${settle_time}
+
+		# Measure network latency
+		if [[ ${nc_reqs} -ge 1 ]]; then
+			mkdir -p "$RESULT_DIR" 2>/dev/null || true
+			local latency_raw_output="$RESULT_DIR/${TEST_NAME// /-}.tmaster_tworker_pods_req_ipaddr_lattot_latconn_latio_latdisconn_rx.raw"
```

Comment:
I still don't like the mega huge raw name.... or that it's not a json or actual csv :-)